### ISSUE TITLE
window.navigator.userAgent cannot be mocked

### ISSUE THREAD

**Issue by elwayman02** [simonihmig/ember-window-mock#585]
Seeing the following error when upgrading from v0.8.1 to v1.0.2:

`Cannot set property userAgent of #<Navigator> which has only a getter`

Here's an example of our code:

```
  setupWindowMock(hooks);

  hooks.beforeEach(function () {
    window.navigator.userAgent = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36';
  });
```

**Comment by elwayman02** [comment]
As a workaround, I found that sinon can be used to get around this issue (though it's not quite ideal, as one would expect `setupWindowMock` to take care of this already).

Here's an example for anyone else having this issue:

`sinon.stub(window.navigator, 'userAgent').get(() => 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36');`

**Comment by simonihmig** [comment]
Hm, that is weird. There is an explicit test for this [here](https://github.com/simonihmig/ember-window-mock/blob/master/test-app/tests/unit/run-window-tests.ts#L378).

Just to make sure, `window` in your example is not using the global but the imported one, right?

**Comment by elwayman02** [comment]
Yes, there's an explicit import and it worked in 0.8.1 but fails in 1.0.2.

**Comment by elwayman02** [comment]
BTW even though I filed this at the same time as #586, they are occurring independently in two separate addons, not the same project. So these issues have been reproducible for me across multiple projects that were already using ember-window-mock 0.8.1 successfully.

**Comment by elwayman02** [comment]
Here's a full look at what I have (this is a test util in an addon's test-support folder):

```
import type { TestContext } from '@ember/test-helpers';
import window from 'ember-window-mock';
import { setupWindowMock } from 'ember-window-mock/test-support';
import sinon from 'sinon';

const MOBILE =
  'Mozilla/5.0 (iPhone; CPU iPhone OS 8_0_2 like Mac OS X) AppleWebKit/600.1.4 (KHTML, like Gecko) Version/8.0 Mobile/12A366 Safari/600.1.4';

const DESKTOP =
  'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.95 Safari/537.36';

interface TestContextExtended extends TestContext {
  setIsMobile(isMobile?: boolean): void;
}

export default function setupIsMobile(hooks: NestedHooks): void {
  setupWindowMock(hooks);

  hooks.beforeEach(function (this: TestContextExtended) {
    this.setIsMobile = (isMobile = true) => {
      if (isMobile) {
        sinon.stub(window.navigator, 'userAgent').get(() => MOBILE);
      } else {
        sinon.stub(window.navigator, 'userAgent').get(() => DESKTOP);
      }
    };
  });
}
```

Package versions:
ember-window-mock v1.0.2
ember-auto-import v2.6.3
ember-source v4.12.1
ember-cli v4.12.2
ember-qunit v8.0.2
@embroider/macros v1.16.5

**Comment by simonihmig** [comment]
> BTW even though I filed this at the same time as https://github.com/simonihmig/ember-window-mock/issues/586, they are occurring independently in two separate addons, not the same project. So these issues have b

Still sounds very much like the same root cause. The only significant change has been the conversion to v2. Which makes ember-auto-import handle the addon. Any chance you could test this with the latest version of that?

**Comment by elwayman02** [comment]
Even with ember-auto-import 2.8.1, the window object does not seem to get proxied by setupWndowMock

**Comment by simonihmig** [comment]
@elwayman02 would you be able to provide a simple reproduction repo? 

Still failing to understand why it would not work for you while it is working in this repo's test-app...
